app/controllers/food_access_points_controller.rb

```diff
@@ -1,9 +1,54 @@
 class FoodAccessPointsController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
   def index
     @food_access_points = FoodAccessPoint.all
   end
 
   def show
     @food_access_point = FoodAccessPoint.find(params[:id])
   end
+  
+  def new
+    @food_access_point = FoodAccessPoint.new
+  end
+
+  def create
+    Rails.logger.debug params.inspect
+    @food_access_point = current_user.food_access_points.build(food_access_point_params(:name, :address, :town, :state, :zip, :phone_number, :website))
+    if @food_access_point.save 
+      redirect_to food_access_point_path(@food_access_point)
+    else
+      render 'edit'
+    end
+  end
+
+  def edit
+    @food_access_point = FoodAccessPoint.find(params[:id])
+  end
+
+  def update
+    @food_access_point = FoodAccessPoint.find(params[:id])
+    @food_access_point.update(food_access_point_params(:name, :address, :town, :state, :zip, :phone_number, :website))
+    if @food_access_point.save
+      redirect_to food_access_point_path(@food_access_point)
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @food_access_point = FoodAccessPoint.find(params[:id])
+    @food_access_point.destroy
+    redirect_to root_path
+  end
```

Comment:
all of these actions look great! i suppose some future feature might restrict deleting food access points to some kind of "super admin" so people don't just go about deleting these by accident or something.

Reply:
same for creating food access points. probably want to restrict that to a super user. but we can come back for that later, doesn't need to be part of this scope of work

Reply:
I was working under the impression that "users" were only people who would be editing/adding access points...staff presumably...so i was thinking more along the lines of approving registrations -not letting people sign up in the app...since that would provide some gatekeeping to only authorized folks

Reply:
ah i see, yes we can figure this out a little later. might be worth getting feedback on this from little rivers